core/src/main/scala/org/apache/spark/rdd/CoalescedRDD.scala

```diff
@@ -219,9 +221,8 @@ private class DefaultPartitionCoalescer(val balanceSlack: Double = 0.10)
    * @param key string representing a partitioned group on preferred machine key
    * @return Option of [[PartitionGroup]] that has least elements for key
    */
-  def getLeastGroupHash(key: String): Option[PartitionGroup] = {
-    groupHash.get(key).map(_.sortWith(compare).head)
-  }
+  def getLeastGroupHash(key: String): Option[PartitionGroup] =
+    groupHash.get(key).flatMap (group => if (group.isEmpty) None else Some(group.min))
```

Comment:
`.map()` should still be OK here as `None` will results in `None`. Rather than the implicit ordering, just use `.minBy`?
`.map(_.minBy(_.numPartitions))`?

Reply:
I'm suggesting you also not make an Option inside the function. You're creating None/Some just to flatten it again.